api/v1/collection.go

```diff
@@ -1,69 +1,80 @@
+	CollectionID          uuid.UUID    `json:"collection_id"`
+	CollectionName        string       `json:"name"`
+	CollectionActive      *bool        `json:"active"`
+	CollectionPieceSource pgtype.JSONB `json:"piece_list_source"`
+	CollectionDealParams  pgtype.JSONB `json:"deal_params"`
 }
 
-type DealParams struct {
-	DurationDays     uint `json:"duration_days"`
-	StartWithinHours uint `json:"start_within_hours"`
+func (a *apiV1) ConfigureCollectionRouter(e *echo.Group) {
+	g := e.Group("/collections")
+	g.GET("", a.handleGetCollections)
+	g.PUT("/:collectionUUID", a.handleModifyCollection)
+	g.POST("", a.handleCreateCollection)
+	g.DELETE("/:collectionUUID", a.handleDeleteCollection)
 }
 
-type Collection struct {
-	UUID uuid.UUID `json:"uuid"`
-	MutableCollection
-}
+// handleCreateCollection godoc
+//
+//	@Summary		Creates a new collection
+//	@Security		apiKey
+//	@Param 			collection CollectionMutable body true "Collection to create"
+//	@Produce		json
+//	@Success		200	{object}	ResponseEnvelope{response=Collection}
+//	@Router			/collections [post]
+func (a *apiV1) handleCreateCollection(c echo.Context) error {
+	var collection db.Collection
 
-type PieceListSource struct {
-	Method            string `json:"method"`
-	PollIntervalHours int    `json:"poll_interval_hours,omitempty"`
-	ConnectionDetails string `json:"connection_details"` // TODO: better types / validation for the connection details
-}
+	err := json.NewDecoder(c.Request().Body).Decode(&collection)
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, CreateErrorResponseEnvelope(c, http.StatusInternalServerError, err.Error()))
+	}
 
-type ReplicationConstraint struct {
-	ConstraintID  int `json:"constraint_id"`
-	ConstraintMax int `json:"constraint_max"`
-}
+	collection.TenantID = db.ID(GetTenantContext(c).TenantID)
+	collection.CollectionID = uuid.New()
+	collection.ReplicationConstraints = []db.ReplicationConstraint{{CollectionID: collection.CollectionID, ConstraintID: 0, ConstraintMax: 10}}
+	active := true
+	collection.CollectionActive = &active
 
-type CreateCollectionResponse struct {
-	CollectionID uint   `json:"collection_id"`
-	Status       string `json:"status"`
-}
+	err = a.db.Create(&collection).Error
 
-// handleCreateCollection godoc
-//
-//		@Summary		Creates a new collection
-//		@Security apiKey
-//	  @Param 			collection body MutableCollection true "Collection to create"
-//		@Produce		json
-//		@Success		200	{object}	ResponseEnvelope{response=Collection}
-//		@Router			/collections [post]
-func handleCreateCollection(c echo.Context) error {
-	return c.JSON(http.StatusNotImplemented, map[string]string{})
-}
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, CreateErrorResponseEnvelope(c, http.StatusInternalServerError, err.Error()))
+	}
 
-type GetCollectionsResponse []Collection
+	return c.JSON(http.StatusOK, CreateSuccessResponseEnvelope(c, "Created collection with id: "+collection.CollectionID.String()))
```

Comment:
i generally prefer returning the newly created `collection` object in its entirety instead of a string - I think you can get this from the `db.Create` call